### DISCUSSION TITLE
Why must x have more than one element in remove_na?

### DISCUSSION THREAD

**Discussion by abarozet** [raphaelvallat/pingouin#287] [Q&A] [answered]
Hi,
I am using function pairwise_ttests on my data and it raises an assertion error because one of my groups has only one element.
Apparently this exception is raised by remove_na.
I do not understand why remove_na has to make this assertion.
Unless I am mistaken, there is not theoretical reason why the t-test should fail when one group has only one element.
Any thoughts?

**Comment by raphaelvallat** [comment, marked as answer]
@abarozet unless you are doing a one-sample T-test (which is not supported by pingouin.pairwise_ttests), you must have at least two valid elements in each of your group otherwise the T value and p-value cannot be calculated.

Example using SciPy:

```python
from scipy.stats import ttest_ind
ttest_ind([1], [2, 3, 4])  # Will return NaN and print a RuntimeWarning
ttest_ind([2, 3, 4], 1)  # Same
ttest_ind(1, 1)  # Same
```

**Comment by raphaelvallat** [comment]
That said, we could remove the assertion check in pingouin.remove_na to avoid raising an error when x is empty or has a single element

```python
pg.remove_na([])  # AssertionError
pg.remove_na(np.array([1]))  # AssertionError
```
